discovery/src/main/java/com/ibm/watson/discovery/v2/Discovery.java

```diff
@@ -0,0 +1,647 @@
+
+  /**
+   * Constructs a new `Discovery` client.
+   *
+   * @param versionDate The version date (yyyy-MM-dd) of the REST API to use. Specifying this value will keep your API
+   *          calls from failing when the service introduces breaking changes.
+   */
+  public Discovery(String versionDate) {
+    this(versionDate, ConfigBasedAuthenticatorFactory.getAuthenticator(SERVICE_NAME));
+  }
+
+  /**
+   * Constructs a new `Discovery` client with the specified Authenticator.
+   *
+   * @param versionDate The version date (yyyy-MM-dd) of the REST API to use. Specifying this value will keep your API
+   *          calls from failing when the service introduces breaking changes.
+   * @param authenticator the Authenticator instance to be configured for this service
+   */
+  public Discovery(String versionDate, Authenticator authenticator) {
+    super(SERVICE_NAME, authenticator);
+    com.ibm.cloud.sdk.core.util.Validator.isTrue((versionDate != null) && !versionDate.isEmpty(),
+        "version cannot be null.");
+    this.versionDate = versionDate;
+  }
+
+  /**
+   * List collections.
+   *
+   * Lists existing collections for the specified project.
+   *
+   * @param listCollectionsOptions the {@link ListCollectionsOptions} containing the options for the call
+   * @return a {@link ServiceCall} with a response type of {@link ListCollectionsResponse}
+   */
+  public ServiceCall<ListCollectionsResponse> listCollections(ListCollectionsOptions listCollectionsOptions) {
+    com.ibm.cloud.sdk.core.util.Validator.notNull(listCollectionsOptions,
+        "listCollectionsOptions cannot be null");
+    String[] pathSegments = { "v2/projects", "collections" };
+    String[] pathParameters = { listCollectionsOptions.projectId() };
+    RequestBuilder builder = RequestBuilder.get(RequestBuilder.constructHttpUrl(getServiceUrl(), pathSegments,
+        pathParameters));
+    builder.query("version", versionDate);
+    Map<String, String> sdkHeaders = SdkCommon.getSdkHeaders("discovery", "v2", "listCollections");
+    for (Entry<String, String> header : sdkHeaders.entrySet()) {
+      builder.header(header.getKey(), header.getValue());
+    }
+    builder.header("Accept", "application/json");
+
+    ResponseConverter<ListCollectionsResponse> responseConverter = ResponseConverterUtils.getValue(
+        new com.google.gson.reflect.TypeToken<ListCollectionsResponse>() {
+        }.getType());
+    return createServiceCall(builder.build(), responseConverter);
+  }
+
+  /**
+   * Query a project.
+   *
+   * By using this method, you can construct queries. For details, see the [Discovery
+   * documentation](https://cloud.ibm.com/docs/services/discovery-data?topic=discovery-data-query-concepts).
+   *
+   * @param queryOptions the {@link QueryOptions} containing the options for the call
+   * @return a {@link ServiceCall} with a response type of {@link QueryResponse}
+   */
+  public ServiceCall<QueryResponse> query(QueryOptions queryOptions) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'query(QueryOptions)' has an NPath complexity of 32768](https://app.codacy.com/app/watson-developer-cloud/java-sdk/pullRequest?prid=4552601)